src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -1090,11 +1100,14 @@ protected static function processCustomAssertion(
                     }
 
                     if ($var_name) {
-                        if ('!' === $assertion->rule[0][0][0]) {
-                            $if_types[$var_name] = [[substr($assertion->rule[0][0], 1)]];
-                        } else {
-                            $if_types[$var_name] = [['!' . $assertion->rule[0][0]]];
-                        }
+                        $if_types[$var_name] = [
+                            array_map(
+                                function ($rule) {
+                                    return $rule[0] === '!' ? (string) substr($rule, 1) : ('!' . $rule);
+                                },
+                                array_values($assertion->rule[0])
+                            )
```

Comment:
I'm actually not sure about this. It's supposed to invert the rule, but it effectively turns `"a" | "b" | "c"` into `!"a" | !"b" | !"c"` rather than `!"a" & !"b" & !"c"`

Reply:
yeah, that's an issue. This will probably lead to weird edge cases